Source/GUI/MemWatcher/Dialogs/DlgAddWatchEntry.cpp

```diff
@@ -424,3 +430,32 @@ MemWatchEntry* DlgAddWatchEntry::stealEntry()
   m_entry = nullptr;
   return entry;
 }
+
+void DlgAddWatchEntry::onPointerOffsetContextMenuRequested(const QPoint& pos)
+{
+  QMenu* contextMenu = new QMenu(this);
+
+  for (int i = 0; i < m_offsetsLayout->rowCount(); i++)
+  {
+    QLabel* lbl = (QLabel*)m_offsetsLayout->itemAtPosition(i, 2)->widget();
+
+    QPoint click_pos = lbl->mapFrom(m_pointerWidget, pos);
+    int xPos = click_pos.x();
+    int yPos = click_pos.y();
+
+    if (0 < yPos && yPos < lbl->height() && 0 < xPos && xPos < lbl->width())
+    {
+      QAction* copyAddr = new QAction(tr("&Copy Address"), this);
+      connect(copyAddr, &QAction::triggered, this, [lbl] {
+        QApplication::clipboard()->setText(lbl->text().mid(4, lbl->text().length() - 4));
+      });
+      contextMenu->addAction(copyAddr);
+      if (lbl->text().contains(QString("??")))
+      {
+        copyAddr->setEnabled(false);
+      }
+      contextMenu->popup(lbl->mapToGlobal(click_pos));
+      break;
+    }
+  }
+}
```

Comment:
Question: Shouldn't we be able to register the context menu to the label directly to avoid the shenanigans with the layout, and all the magic numbers and magic strings?

Suggestion:

When the label widgets are inserted in the layout, a Qt property (see [`QObject::setProperty()`](https://doc.qt.io/qt-6/qobject.html#setProperty)) would be set to hold the required information (perhaps an index, or perhaps the memory offset itself). Then, when the label widget is right-clicked, and the context menu is shown, the slot would check the `sender()`'s property (see [`QObject::property()`](https://doc.qt.io/qt-6/qobject.html#property)).

Reply:
I didn't realize that I could get the sender. This is definitely a better way.